test/framework/clusterctl/repository.go

```diff
@@ -168,3 +168,33 @@ func YAMLForComponentSource(ctx context.Context, source ProviderVersionSource) (
 
 	return data, nil
 }
+
+// getComponentSourceFromURL fetches contents of component source YAML file from provided URL source.
+func getComponentSourceFromURL(source ProviderVersionSource) ([]byte, error) {
+	var buf []byte
+
+	u, err := url.Parse(source.Value)
+	if err != nil {
+		return nil, err
+	}
+
+	switch u.Scheme {
+	case fileURIScheme:
+		buf, err = ioutil.ReadFile(u.Path)
+		if err != nil {
+			return nil, err
+		}
+	default:
```

Comment:
Would it be better to be explicit and return an error for when there is an unknown scheme? This would mean users would have a better idea of what's going wrong compared with a `http.Get` error which may be cryptic? 

If it were me I would do something like:

```
switch strings.ToLower(u.Scheme){
case "file":
  ...
case "http", "https":
 ...
default:
  return nil, fmt.Errorf("unknown scheme for component source %q: allowed values are file, http, https", u.Scheme)
}
```

WDYT?

I guess this does leave a question open if the user doesn't specify the scheme, how do we handle this, http?

Reply:
We can follow the way `--file` flag to `kubectl apply` takes input.
By default, it interprets the value passed to that flag as a file, and if the value is a url then (with `http://` or `https://` prefix), then use the http URL.

Reply:
Being consistent would be great, but would that be a breaking change to the existing functionality? 
Has anyone tried doing this and not putting a scheme and seen the result? Perhaps if we can verify what the current behaviour is that will help to determine what should be done here

Reply:
@JoelSpeed Just checked, if no scheme is provided, `http.Get` returns error `"unsupported protocol scheme \"\""`. So i guess, interpreting value without scheme as flag would be ok.

Reply:
@JoelSpeed Updated code, now empty scheme interpreted as file.

Reply:
That's what I expected, thanks for confirming.